mlrun/model_monitoring/db/stores/__init__.py

```diff
@@ -95,24 +87,40 @@ def get_model_endpoint_store(
 
 def get_store_object(
     project: str,
-    access_key: str = None,
+    store_type: typing.Optional[str] = None,
     secret_provider: typing.Callable = None,
+    **kwargs,
 ) -> StoreBase:
     """
     Getting the DB target type based on mlrun.config.model_endpoint_monitoring.store_type.
 
     :param project:         The name of the project.
-    :param access_key:      Access key with permission to the DB table.
+    :param store_type:      The type of the store target. See mlrun.model_monitoring.db.stores.ObjectStoreFactory
+                            for available store types.
     :param secret_provider: An optional secret provider to get the connection string secret.
 
     :return: `StoreBase` object. Using this object, the user can apply different operations on the
              model monitoring record such as write, update, get and delete a model endpoint.
     """
 
+    store_connection_string = mlrun.model_monitoring.helpers.get_connection_string(
+        secret_provider=secret_provider
+    )
+
+    if store_connection_string and (
+        store_connection_string.startswith("mysql")
+        or store_connection_string.startswith("sqlite")
+    ):
```

Comment:
To resolve the previous comment do this only if the `store_type` is `None` or SQL.

Reply:
but that's the thing - the store type should be decided based on the connection string